### PR TITLE
Pytest

### PR DESCRIPTION
**Pull request by Jokeswar**
I believe we might want to break this into separate tests. Also it uses this version of `vagrant-vmck` https://github.com/vmck/vagrant-vmck/blob/vagrant-pytest/submission/launch.sh (vagrant has some trouble installing rsync in a docker container). Is there anything we might want to test?

Fixes https://github.com/vmck/acs-interface/issues/87

### CODE REVIEW

File: testsuite/test_interface.py [hunk 1]
```diff
@@ -30,16 +38,35 @@ def test_submission(client):
...
     while submission.state != submission.STATE_DONE:
-        time.sleep(2)
+        submission = Submission.objects.all()[0]
 
-    assert submission.score > 0
+        time.sleep(10)
+
+        count += 1
+        if count == 18:
+            assert 0 == 1
```
Comment (gmuraru) [inline testsuite/test_interface.py]: ```assert False```

File: testsuite/test_interface.py [hunk 2]
```diff
@@ -30,16 +38,35 @@ def test_submission(client):
         )
 
     assert len(Submission.objects.all()) == 1
-    assert storage.exists('1.zip')
+    assert storage.exists('1.zip') is True
```
Comment (gmuraru) [inline testsuite/test_interface.py]: Q: Shouldn't be ok only with ```assert storage.exists(`1.zip`)```?

File: ci/test-pytest.sh [hunk 3]
```diff
@@ -1,4 +1,16 @@
...
+MINIO_SECRET_KEY=123456789
+VMCK_API_URL=http://10.66.60.1:10000/v0/
+DEBUG=True
+HOSTNAME=*
+ACS_INTERFACE_ADDRESS=http://10.66.60.1:8000/
+MANAGER_TAG=vagrant-pytest" > .env
+
+pipenv run pytest --liveserver 10.66.60.1:8000
+
+rm .env
```
Comment (gmuraru) [inline ci/test-pytest.sh]: Q: ```rm -rf```?

File: testsuite/test_interface.py (hunk 2 above)
Comment (Jokeswar) [reply testsuite/test_interface.py]: Good point. I guess I am stuck with a bit of verbosity from C

File: ci/test-pytest.sh (hunk 3 above)
Comment (Jokeswar) [reply ci/test-pytest.sh]: Actually we might not even need this. It's only run on the CI so we don't care about clean-up.

File: testsuite/test_interface.py [hunk 4]
```diff
@@ -35,11 +43,30 @@ def test_submission(client):
...
+    assert submission.score == 100
+    assert submission.total_score == 100
+
+    client.post(
+        '/submission/1/review',
+        data={'review-code': '+10.0: Good Job'},
+        HTTP_REFERER='/'
+    )
+
+    submission = Submission.objects.all()[0]
```
Comment (gmuraru) [inline testsuite/test_interface.py]: Maybe add an assert for ```objects.all()``` - that it has ```len 1```

File: testsuite/test_interface.py [hunk 5]
```diff
@@ -35,11 +43,30 @@ def test_submission(client):
...
     assert submission.state == submission.STATE_NEW
-    assert submission.assignment.code == 'pc-a0'
+    assert submission.assignment.code == 'a0'
     assert submission.archive_size > 0
     assert submission.vmck_job_id > 0
 
+    count = 0
     while submission.state != submission.STATE_DONE:
-        time.sleep(2)
+        submission = Submission.objects.all()[0]
```
Comment (mgax) [inline testsuite/test_interface.py]: You can call [refresh_from_db](https://docs.djangoproject.com/en/3.0/ref/models/instances/#django.db.models.Model.refresh_from_db) to update the model instance.

File: testsuite/test_interface.py [hunk 6]
```diff
@@ -35,11 +43,30 @@ def test_submission(client):
...
 
-    assert submission.score > 0
+        time.sleep(10)
```
Comment (mgax) [inline testsuite/test_interface.py]: Let's go with a shorter sleep, even 1s should be fine.

File: testsuite/test_interface.py [hunk 7]
```diff
@@ -35,11 +43,30 @@ def test_submission(client):
...
+
+        count += 1
+        if count == 18:
```
Comment (mgax) [inline testsuite/test_interface.py]: I suggest you use [`time()`](https://docs.python.org/3/library/time.html#time.time) to decide when to bail from the loop, it's easier to understand when reading the code.

File: testsuite/test_interface.py [hunk 8]
```diff
@@ -35,11 +43,30 @@ def test_submission(client):
...
+            assert False
+
+    assert submission.score == 100
+    assert submission.total_score == 100
+
+    client.post(
+        '/submission/1/review',
+        data={'review-code': '+10.0: Good Job'},
+        HTTP_REFERER='/'
```
Comment (mgax) [inline testsuite/test_interface.py]: Final comma pls :)

**Pull request merged**